### DISCUSSION TITLE
Install of Sabnzbd not working

### DISCUSSION THREAD

**Discussion by DaulRa** [tteck/Proxmox#3333] [Help Needed]
Seems to time out at this phase while installing SABnzbd:

SABnzbd-4.3.2/interfaces/wizard/one.html
SABnzbd-4.3.2/interfaces/wizard/README.TXT
SABnzbd-4.3.2/interfaces/wizard/static/style.css
SABnzbd-4.3.2/interfaces/wizard/static/javascript/checkserver.js
 /Ignoring pywin32: markers 'sys_platform == "win32"' don't match your environment
Ignoring windows-toasts: markers 'sys_platform == "win32" and python_version > "3.8"' don't match your environment
Ignoring winrt-runtime: markers 'sys_platform == "win32" and python_version > "3.8"' don't match your environment
Ignoring winrt-Windows.Data.Xml.Dom: markers 'sys_platform == "win32" and python_version > "3.8"' don't match your environment
Ignoring winrt-Windows.Foundation: markers 'sys_platform == "win32" and python_version > "3.8"' don't match your environment
Ignoring winrt-Windows.Foundation.Collections: markers 'sys_platform == "win32" and python_version > "3.8"' don't match your environment
Ignoring winrt-Windows.UI.Notifications: markers 'sys_platform == "win32" and python_version > "3.8"' don't match your environment
Ignoring pyobjc-core: markers 'sys_platform == "darwin"' don't match your environment
Ignoring pyobjc-framework-Cocoa: markers 'sys_platform == "darwin"' don't match your environment
 -Collecting apprise==1.8.0
 \  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x73358877ed50>: Failed to establish a new connection: [Errno -2] Name or service not known')': /packages/cd/1e/a7ba1c528ae9e3411e6f76c05aefba4e08feac4f94742de4bf1f0085d844/apprise-1.8.0-py3-none-any.whl
 \  WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x73358877f350>: Failed to establish a new connection: [Errno -2] Name or service not known')': /packages/cd/1e/a7ba1c528ae9e3411e6f76c05aefba4e08feac4f94742de4bf1f0085d844/apprise-1.8.0-py3-none-any.whl
 \  WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x73358877fbd0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /packages/cd/1e/a7ba1c528ae9e3411e6f76c05aefba4e08feac4f94742de4bf1f0085d844/apprise-1.8.0-py3-none-any.whl
 \  WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x733588754450>: Failed to establish a new connection: [Errno -2] Name or service not known')': /packages/cd/1e/a7ba1c528ae9e3411e6f76c05aefba4e08feac4f94742de4bf1f0085d844/apprise-1.8.0-py3-none-any.whl
 |  WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x73358868c5d0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /packages/cd/1e/a7ba1c528ae9e3411e6f76c05aefba4e08feac4f94742de4bf1f0085d844/apprise-1.8.0-py3-none-any.whl
 \ERROR: Could not install packages due to an OSError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Max retries exceeded with url: /packages/cd/1e/a7ba1c528ae9e3411e6f76c05aefba4e08feac4f94742de4bf1f0085d844/apprise-1.8.0-py3-none-any.whl (Caused by NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x73358868cd10>: Failed to establish a new connection: [Errno -2] Name or service not known'))

[ERROR] in line 43: exit code 0: while executing command $STD python3 -m pip install -r requirements.txt

This happens both in Advanced and Default install modes.

**Comment by tteck** [comment]
🏁 The script runs without any problems from this end.

```
   _____ ___    ____              __        __
  / ___//   |  / __ )____  ____  / /_  ____/ /
  \__ \/ /| | / __  / __ \/_  / / __ \/ __  / 
 ___/ / ___ |/ /_/ / / / / / /_/ /_/ / /_/ /  
/____/_/  |_/_____/_/ /_/ /___/_.___/\__,_/   
                                              
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 112
Using Hostname: sabnzbd
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a SABnzbd LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 112 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.112.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Updated Python3
 ✓ Installed SABnzbd
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

SABnzbd should be reachable by going to the following URL.
         http://192.168.86.250:7777 
```

**Comment by CameronAM** [comment]
This was breaking for me just now when attempting to _upgrade_ sabnzbd. Python3 is complaining that it would not install dependencies without a venv configured.

I assume something similar is happening on installation.

For updating, this fixed it:

```
apt install python3-venv
cd /opt/sabnzbd
python3 -m venv venv
./venv/bin/python3 -m pip install -r requirements.txt
```

Then modify `/etc/systemd/system/sabznbd.service` so the `ExecStart` line is `/opt/sabnzbd/venv/bin/python3 SABnzbd.py -s 0.0.0.0:7777`

Then run:

```
systemctl daemon-reload
service sabnzbd start
```

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/3300
